fluent/protocol/handshake.go

```diff
@@ -158,7 +159,10 @@ func validateDigest(received, key, nonce, salt []byte, hostname string) error {
 func computeHexDigest(salt []byte, hostname string, nonce, sharedKey []byte) []byte {
 	h := sha512.New()
 	h.Write(salt)
-	io.WriteString(h, hostname)
+	_, err := io.WriteString(h, hostname)
```

Comment:
I think this function needs to be changed to return an error. Should be simple since validateDigest already returns one.